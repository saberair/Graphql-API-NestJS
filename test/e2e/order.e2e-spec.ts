import { getApolloServer } from '@nestjs/apollo';
import gql from 'graphql-tag';
import { CreateOrderInput } from '../../src/order/dto/create-order.input';
import { AppFixture, createAppFixture } from './e2e-setup';

describe('Order (e2e)', () => {
  let appFixture: AppFixture;
  let apolloServer: ReturnType<typeof getApolloServer>;

  afterAll(async () => {
    appFixture.destroy();
  });

  beforeAll(async () => {
    appFixture = await createAppFixture();
    apolloServer = appFixture.apolloServer;
  });

  beforeEach(async () => {
    await appFixture.reset();
  });

  it('creates a new order', async () => {
    const Create_Order_Mutation = gql`
      mutation createOrder($createOrderInput: CreateOrderInput!) {
        createOrder(createOrderInput: $createOrderInput) {
          id
          quantity
          status
          coffee {
            id
            name
          }
        }
      }
    `;

    const createOrderInput: CreateOrderInput = {
      coffeeId: 1,
      quantity: 5,
      status: 'Done',
    };

    const response = await apolloServer.executeOperation({
      query: Create_Order_Mutation,
      variables: { createOrderInput },
    });

    expect(response).toMatchSnapshot({
      body: {
        singleResult: {
          data: {
            createOrder: {
              id: expect.any(Number),
            },
          },
        },
      },
    });
  });

  it('finds all orders', async () => {
    const Find_All_Orders_Query = gql`
      query FindAllOrders {
        findAllOrders {
          id
          quantity
          status
          coffee {
            id
            name
          }
        }
      }
    `;

    const response = await apolloServer.executeOperation({
      query: Find_All_Orders_Query,
    });

    expect(response).toMatchSnapshot();
  });
});
