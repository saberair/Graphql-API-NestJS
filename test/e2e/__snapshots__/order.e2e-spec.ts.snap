// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order (e2e) creates a new order 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "createOrder": Object {
          "coffee": Object {
            "id": 1,
            "name": "Cappucino",
          },
          "id": Any<Number>,
          "quantity": 5,
          "status": "Done",
        },
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;

exports[`Order (e2e) finds all orders 1`] = `
Object {
  "body": Object {
    "kind": "single",
    "singleResult": Object {
      "data": Object {
        "findAllOrders": Array [
          Object {
            "coffee": Object {
              "id": 1,
              "name": "Cappucino",
            },
            "id": 1,
            "quantity": 2,
            "status": "InProgress",
          },
          Object {
            "coffee": Object {
              "id": 1,
              "name": "Cappucino",
            },
            "id": 2,
            "quantity": 1,
            "status": "Done",
          },
        ],
      },
      "errors": undefined,
    },
  },
  "http": Object {
    "headers": Map {
      "cache-control" => "no-store",
    },
    "status": undefined,
  },
}
`;
